docs/proposals/disk-affinity.md

```diff
@@ -0,0 +1,161 @@
+# Proposal: Supporting disk affinity/anti-affinity feature 
+
+Last updated: 18.01.2022
+
+## Abstract
+
+This proposal contains approach for supporting disk affinity/anti-affinity feature.
+
+## Background
+
+Current lvg storageclass cannot mount physical disks to PVs with the awareness of the requesting pods. So, it's possible
+for different applications to share the same disk.
+CSI should provide a way to support:
+* mount PVs for the same pod on the same disk (disk affinity);
+* do not mount PVs on the disks that already have some PVs from other specific pods (disk antiaffinity).
+
+## Proposal
+
+Usually for affinity/antiaffinity topology feature is used. But the minimal domain, on which this feature operates, is nodes, not disks.
```

Comment:
Did you check other CSI drivers implementations? Especially vSAN CSI? Do they have similar logic?

Reply:
About vSAN CSI. All what I found up to now are the following annotations, which are placed directly at PVCs:
```
// best effort to spread PVs that have “foo-inst-6” as annotation on different disks. 
placement.beta.vmware.com/storagepool_antiAffinityPrefered : foo-inst-6 
```
The preferred policy asks for a best-effort anti-affinity, which means as far as possible all PVCs that have the annotation “placement.beta.vmware.com/storagepool_antiAffinityPrefered : foo-inst-6” should be placed on differnt vsan-direct disks, but PV creation should not fail if this anit-affinity can not be satisfied.
```
// make sure no to PVs with same PVC annotation go to the same disk
placement.beta.vmware.com/storagepool_antiAffinityRequired : foo-inst-6
```
If storagepool_antiAffinityRequired is used, then every PV with PVC annotation “placement.beta.vmware.com/storagepool_antiAffinityRequired : foo-inst-8” has to be placed on a different vSAN direct disk. Note that the node for the PV placement is still governed by where the corresponding POD gets placed, this anti-affinity rule only applies for PVs going to the same node.

Reply:
topolvm hasn't this feature
direct-csi hasn't this feature
pmem-csi hasn't this feature
csi-driver-lvm hasn't this feature